src/System.Text.Json/tests/Serialization/CustomConverterTests.BadConverters.cs

```diff
@@ -76,8 +77,102 @@ private class PocoWithNullConverter
+            {
+                Assert.Contains("'System.Text.Json.Serialization.Tests.CustomConverterTests+PocoWithNullConverter.MyInt'", ex.Message);
+            }
+        }
+
+        private class InvalidTypeConverterClass
+        {
+            [JsonConverter(typeof(JsonStringEnumConverter))]
+            public ICollection<InvalidTypeConverterEnum> MyEnumValues { get; set; }
+        }
+
+        private enum InvalidTypeConverterEnum
+        {
+            Value1,
+            Value2,
+        }
+
+        [Fact]
+        public static void AttributeOnPropertyFail()
+        {
+            try
+            {
+                JsonSerializer.Serialize(new InvalidTypeConverterClass());
+                Assert.True(false, "Expected InvalidOperationException was not thrown.");
+            }
+            catch (InvalidOperationException ex)
+            {
+                // Message should be in the form "The converter specified on 'System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterClass.MyEnumValues' is not compatible with the type 'System.Collections.Generic.ICollection`1[System.Text.Json.Serialization.Tests.ExceptionTests+InvalidTypeConverterEnum]'."
+                Assert.Contains("'System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterClass.MyEnumValues'", ex.Message);
+                Assert.Contains("'System.Collections.Generic.ICollection`1[System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterEnum]'", ex.Message);
+            }
+
+            try
+            {
+                JsonSerializer.Deserialize<InvalidTypeConverterClass>("{}");
+                Assert.True(false, "Expected InvalidOperationException was not thrown.");
+            }
+            catch (InvalidOperationException ex)
+            {
+                Assert.Contains("'System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterClass.MyEnumValues'", ex.Message);
+                Assert.Contains("'System.Collections.Generic.ICollection`1[System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterEnum]'", ex.Message);
+            }
+        }
+
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        private class InvalidTypeConverterClassWithAttribute { }
+
+        [Fact]
+        public static void AttributeOnClassFail()
+        {
+            try
+            {
+                JsonSerializer.Serialize(new InvalidTypeConverterClassWithAttribute());
+                Assert.True(false, "Expected InvalidOperationException was not thrown.");
+            }
+            catch (InvalidOperationException ex)
+            {
+                // Message should be in the form "The converter specified on 'System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterClassWithAttribute' is not compatible with the type 'System.Text.Json.Serialization.Tests.ExceptionTests+InvalidTypeConverterClassWithAttribute'."
```

Comment:
Is there a comment mismatch?
Where is "ExceptionTests+InvalidTypeConverterClassWithAttribute" coming from?

Reply:
Thanks yes updated that comment.